ddtrace/internal/ci_visibility/recorder.py

```diff
@@ -72,6 +77,41 @@ def __init__(self, tracer=None, config=None, service=None):
         except Exception:
             log.warning("Failed to load CODEOWNERS", exc_info=True)
 
+    def _configure_writer(self):
+        writer = None
+        if ddconfig._ci_visibility_agentless_enabled:
+            headers = {"dd-api-key": os.environ.get("DD_API_KEY")}
+            if headers["dd-api-key"]:
+                writer = CIVisibilityWriter(
+                    intake_url="%s.%s" % (AGENTLESS_BASE_URL, os.environ.get("DD_SITE", AGENTLESS_DEFAULT_SITE)),
+                    headers=headers,
+                )
+            else:
+                raise EnvironmentError(
+                    "DD_CIVISIBILITY_AGENTLESS_ENABLED is set, but DD_API_KEY is not set, so ddtrace "
+                    "cannot be initialized."
+                )
+        elif self._agent_evp_proxy_is_available():
```

Comment:
Do we need a final `else` case where we fall back to serialising to regular APM traces (ie; the behaviour of the currently released tracer), or will that happen automatically?

Reply:
I believe this conflicts with @juan-fernandez's comment here: https://github.com/DataDog/dd-trace-py/pull/5497#discussion_r1171082980

Reply:
I mean when `DD_CIVISIBILITY_AGENTLESS_ENABLED` is not set. First we should try to use the EVP proxy, but if it is not available, fall back to sending just test traces.

Reply:
Pseudocode:
```
if DD_CIVISIBILITY_AGENTLESS_ENABLED:
    if not DD_APIKEY:
        throw
    else:
        writer = AgentlessWriter
elif evp_proxy_available:
    writer = EvpProxyWriter
else:
    writer = ApmWriter
```